### DISCUSSION TITLE
Multiple @next/font fonts

### DISCUSSION THREAD

**Discussion by wiesson** [vercel/next.js#41872] [Help] [answered]
I have two fonts (headlines and text) that I'd like to include. Does anyone know how that might work? It looks like that the classNames are overwriting each other.

```Javascript
import {
  Playfair_Display as PlayfairDisplay,
  Raleway,
} from "@next/font/google";

const playfairDisplay = PlayfairDisplay();
const raleway = Raleway();

<html className={`${playfairDisplay.className} ${raleway.className}`}>
// ...
</html>
```

**Comment by hanneslund** [comment, marked as answer]
In the snippet you're adding both font families to the `<html>`.
```css
/* End result would be something like this */
html {
  font-family: playfair;
  font-family: raleway;
}
```

If you want to use Raleway on all your texts, you can apply it to `<html>`.
```tsx
// app/layout.tsx - root layout
import { Raleway } from "@next/font/google";

const raleway = Raleway();

<html className={raleway.className}>
// ...
</html>
```

And then use Playfair Display on headers
```tsx
// fonts.ts
import { Playfair_Display } from "@next/font/google";

export const playfairDisplay = Playfair_Display();


// some layout/page/component
import { playfairDisplay } from "./fonts"

<h1 className={playfairDisplay.className}>
```
Or some of the other ways to [apply styles](https://beta.nextjs.org/docs/api-reference/components/font#applying-styles).

**Reply by wiesson** [reply]
Thank you! And how do I use it with tailwind or global css?

**Reply by SimonVillage** [reply]
Seems like that in the current state it is not possible to use it with [prose](https://tailwindcss.com/docs/typography-plugin) from tailwind as `font-weight` is defined in the tailwind classes.

**Reply by pruhstal** [reply]
@wiesson try wrapping your app with:

  ```<main className={`${raleway.className} ${playfairDisplay.variable}`}>```

Yes, you're reading that correctly -- combining `className` and `variable` seems to be the hack for getting two separate fonts loading using `next/font` with tailwind.

**Reply by mailmonir** [reply]
where to create font.js file? when creating font.js file in root or other folder exporting font from there says undefined

**Comment by gvocale** [comment]
Also how to handle normal vs italic?

For example Work Sans variable font. This doesn't seem to work:

```tsx
const workSansNormal = Work_Sans({
  fallback: ["sans-serif"],
  style: "normal",
  variable: "--font-work-sans",
  adjustFontFallback: true,
});

const workSansItalic = Work_Sans({
  fallback: ["sans-serif"],
  style: "italic",
  variable: "--font-work-sans",
  adjustFontFallback: true,
});
```

```tsx
export default function RootLayout({ children }: { children: ReactNode }) {
  return (
    <html
      lang="en"
      className={`${feijoaMedium.variable} ${workSansNormal.variable} ${workSansItalic.variable}`}
    >
      <body>{children}</body>
    </html>
  );
}
```

When inspecting seems that both fonts are using the same variable, but not being assigned respectively to style normal vs italic

<img width="366" alt="Screen Shot 2022-10-28 at 2 51 58 PM" src="https://user-images.githubusercontent.com/110624927/198721239-8cfc122f-f787-41c6-91e5-504de62a4b11.png">

**Reply by hanneslund** [reply]
This is possible in `13.0.1` https://github.com/vercel/next.js/issues/41969#issuecomment-1298314889

```tsx
const workSans = Work_Sans({
  style: ['normal', 'italic'],
  variable: '--font-work-sans',
  subsets: ['latin'],
})
```
